auth/store.go

```diff
@@ -361,6 +378,70 @@ func (as *authStore) RoleGrant(r *pb.AuthRoleGrantRequest) (*pb.AuthRoleGrantRes
 	return &pb.AuthRoleGrantResponse{}, nil
 }
 
+func (as *authStore) isOpPermitted(userName string, key string, write bool, read bool) bool {
+	if !as.enabled {
+		return true
+	}
+
+	tx := as.be.BatchTx()
+	tx.Lock()
+	defer tx.Unlock()
+
+	_, vs := tx.UnsafeRange(authUsersBucketName, []byte(userName), nil, 0)
```

Comment:
this seems to be very expensive. we probably need to do some caching in the future.